fancyerror.go

```diff
@@ -129,7 +130,21 @@ var maxErrorBodySize = 200 * 1024
 func isJSONMediaType(header http.Header) bool {
 	contentType := header.Get("Content-Type")
 	mediaType, _, _ := mime.ParseMediaType(contentType)
-	return mediaType == "application/json"
+	m := strings.TrimPrefix(mediaType, "application/")
+	if len(m) == len(mediaType) {
+		return false
+	}
+	// Look for +json suffix. See https://tools.ietf.org/html/rfc6838#section-4.2.8
+	for {
```

Comment:
It took me a little while to fathom what is going on here. perhaps add a comment saying exactly what we're looking for.

I wonder if
```
for _, p := range strings.Split(mediaType, "+") {
    if p == "json" {
        return true
    }
}

return false
```
is significantly less efficient. I guess you want to avoid allocation if you can.

Reply:
Yeah, I'd prefer to avoid the slice allocation. I'll add a comment.